scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -224,3 +277,155 @@ def _calculate_sizes(self):
             self.sections.append({"name": name, "load_addr": load_addr,
                                   "size": size, "virt_addr": virt_addr,
                                   "type": stype, "recognized": recognized})
+
+    def _get_footprint_from_elf_file(self) -> None:
+        self._check_elf_file()
+        self._check_is_xip()
+        self._calculate_from_elf_sections()
+
+    def _get_buildlog_file_content(self) -> typing.List[str]:
+        """Get content of the build.log file.
+
+        @return Content of the build.log file (typing.List[str])
+        """
+        if os.path.exists(path=self.filename) and self.filename.endswith("build.log"):
+            with open(file=self.filename, mode='r') as file:
+                content_file = file.readlines()
+        else:
+            logger.error("Incorrect build.log file to analyse of the footprints.")
+            content_file = []
+        return content_file
+
+    def _search_idx_of_last_pattern_occurrence(self, file_content: typing.List[str], pattern: str) -> int:
+        """Search number of line with information about memory footprint.
+
+        @param file_content (str) Content of build.log.
+        @param pattern (str) String with start phrase of memory footprint section.
+        @return: Number of offset with information about the memory footprint (int)
+        """
+        result = 0
+        if not pattern:
+            pattern = self.PATTERN_SEARCHED_LINE
+
+        # Check file from tail
+        for idx, line in enumerate(reversed(file_content)):
+            # Condition is fullfied, if pattern is matched with start of line.
+            if re.match(pattern=pattern, string=line):
+                result = idx
+                break
+        # If file does not contain information about memory footprint, warning is raised.
+        if result == 0:
+            logger.error(msg=self.MSG_NOT_FOUND_FOOTPRINT)
+        return result
```

Comment:
Please unify wording in comments and variables names: offset, number of a line, index etc.

Reply:
Done